akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MiscDirectivesSpec.scala

```diff
@@ -65,6 +65,72 @@ class MiscDirectivesSpec extends RoutingSpec {
     }
   }
 
+  "the withSizeLimit directive" should {
+    "not apply if entity is not consumed" in {
+      val route = withSizeLimit(500) { completeOk }
+
+      Post("/abc", entityOfSize(500)) ~> route ~> check {
+        status shouldEqual StatusCodes.OK
+      }
+
+      Post("/abc", entityOfSize(501)) ~> route ~> check {
+        status shouldEqual StatusCodes.OK
+      }
+    }
+
+    "apply if entity is consumed" in {
+      val route = withSizeLimit(500) {
+        entity(as[String]) { _ ⇒
+          completeOk
+        }
+      }
+
+      Post("/abc", entityOfSize(500)) ~> route ~> check {
+        status shouldEqual StatusCodes.OK
+      }
+
+      Post("/abc", entityOfSize(501)) ~> Route.seal(route) ~> check {
+        status shouldEqual StatusCodes.BadRequest
+      }
+    }
+
+    "properly handle nested directives" in {
+      val route =
+        withSizeLimit(500) {
+          withSizeLimit(800) {
+            entity(as[String]) { _ ⇒
+              completeOk
+            }
+          }
+        }
+
+      Post("/abc", entityOfSize(800)) ~> route ~> check {
+        status shouldEqual StatusCodes.OK
+      }
+
+      Post("/abc", entityOfSize(801)) ~> Route.seal(route) ~> check {
+        status shouldEqual StatusCodes.BadRequest
+      }
+    }
+  }
+
+  "the withoutSizeLimit directive" should {
```

Comment:
Also I was trying to add test like this:

```
  "the withoutSizeLimit directive" should {
    "accept entities bigger than configured with akka.http.parsing.max-content-length" in {
      val standardRoute =
        entity(as[String]) { _ ⇒
          completeOk
        }

      val unlimitedRoute =
        withoutSizeLimit {
          entity(as[String]) { _ ⇒
            completeOk
          }
        }

      val maxLength = testConfig.getBytes("akka.http.parsing.max-content-length").toInt
      val testRequest = Post("/abc", entityOfSize(maxLength + 100))

      testRequest ~> Route.seal(standardRoute) ~> check {
        status shouldEqual StatusCodes.BadRequest // we expect result to be `BadRequest` but actually `status shouldEqual StatusCodes.OK` held
      }

      testRequest ~> unlimitedRoute ~> check {
        status shouldEqual StatusCodes.OK
      }
    }
  }
```

But seemingly for tests little bit different machinery is invoked - e.g. `PrepareRequests.onPush` is not called in tests - which means that `akka.http.parsing.max-content-length` is not applied in tests. Therefore in tests all routes behaves as they are `withoutSizeLimit`.